pkg/controllers/clusterresourceplacementeviction/controller.go

```diff
@@ -39,36 +43,58 @@ func (r *Reconciler) Reconcile(ctx context.Context, req runtime.Request) (runtim
 	startTime := time.Now()
 	evictionName := req.NamespacedName.Name
 	klog.V(2).InfoS("ClusterResourcePlacementEviction reconciliation starts", "clusterResourcePlacementEviction", evictionName)
+	var internalError bool
 	defer func() {
+		if internalError {
+			metrics.FleetEvictionStatus.WithLabelValues(evictionName, "false").SetToCurrentTime()
+		}
 		latency := time.Since(startTime).Milliseconds()
 		klog.V(2).InfoS("ClusterResourcePlacementEviction reconciliation ends", "clusterResourcePlacementEviction", evictionName, "latency", latency)
 	}()
 
 	var eviction placementv1beta1.ClusterResourcePlacementEviction
 	if err := r.Client.Get(ctx, req.NamespacedName, &eviction); err != nil {
+		internalError = true
 		klog.ErrorS(err, "Failed to get cluster resource placement eviction", "clusterResourcePlacementEviction", evictionName)
 		return runtime.Result{}, client.IgnoreNotFound(err)
 	}
 
 	if isEvictionInTerminalState(&eviction) {
```

Comment:
if the previous update succeeded then why would we reconcile again to come here?

Reply:
Due to the status update on eviciton we reconile again, but moved the metric emission to when we originally update status instead